packages/design-system/src/scalars/components/number-field/number-field-validations.ts

```diff
@@ -31,67 +31,70 @@ export const validateNumericType =
 
     switch (numericType) {
       case "PositiveInt": {
-        if (!isInt) return "Value must be a integer";
+        if (!isInt) return "Value must be a positive integer";
         if (allowNegative) return true;
-        // Si llegue hasta aqui no permito negative
         if (isPositive) return true;
         return "Value must be a positive integer";
       }
 
       case "NegativeInt": {
-        if (!isInt) return "Value must be an integer";
+        if (!isInt) return "Value must be a negative integer";
         if (!allowNegative && Number(value) >= 0)
           return "Value must be a negative integer";
         if (isNegative) return true;
         return "Value must be a negative integer";
       }
 
       case "NonNegativeInt": {
-        if (!isInt) return "Value must be an integer";
+        if (!isInt) return "Value must be a non-negative integer";
         if (!allowNegative && isNegative)
           return "Value must be a non-negative integer";
         if (Number(value) >= 0) return true;
         return "Value must be a non-negative integer";
       }
 
       case "NonPositiveInt": {
-        if (!isInt) return "Value must be an integer";
+        if (!isInt) return "Value must be a non-positive integer";
         if (!allowNegative && Number(value) > 0)
           return "Value must be a non-positive integer";
         if (Number(value) <= 0) return true;
         return "Value must be a non-positive integer";
       }
 
       case "PositiveFloat": {
-        if (!isFloat(value)) return "Value must be a float";
+        if (!isFloat(value) && !isInt)
```

Comment:
`!isFloat(value) && !isInt` this condition is wrong:

Here is the decalation of those functions:

```
export const isInteger = (value: unknown): boolean =>
  Number.isInteger(Number(value));
export const isFloat = (value: unknown): boolean =>
  !Number.isInteger(Number(value));
```

So if we analyze them logically: let's say
isInteger = `I`
isFloat = `F` = `¬I`

So you condition is `¬F ∧ ¬I` => `¬(¬I) ∧ ¬I` => `I  ∧ ¬I`.

This is always going to be false because you're saying it can't be an integer and need to be integer at the same time.

Take in count that for float numbers, you only need to check if it is positive or negative, no if they're int or float, because if they're valid numbers they're going to be float as Integers are also Float

Reply:
you are right , done